src/event-stream/tokens-contract-handler.ts

```diff
@@ -626,27 +648,76 @@ class TokensContractHandler {
   /**
    * Make readonly contract call
    */
-  private async makeReadOnlyContractCall(
+  async makeReadOnlyContractCall(
+    functionName: string,
+    functionArgs: ClarityValue[]
+  ): Promise<ReadOnlyContractCallResponse> {
+    const url = this.stacksNetwork.getReadOnlyFunctionCallApiUrl(
+      this.contractAddress,
+      this.contractName,
+      functionName
+    );
+    const args = functionArgs.map(arg => cvToHex(arg));
+
+    const body = JSON.stringify({
+      sender: this.address,
+      arguments: args,
+    });
+
+    let response: Response;
+
+    try {
+      response = await fetchPrivate(url, {
+        method: 'POST',
+        body,
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      });
+    } catch {
+      // In case there is a network error during fetch.
+      return { found: false, retriable: true };
+    }
+
+    if (!response.ok) {
+      if (response.status >= 500) return { found: false, retriable: true };
+      const msg = await response.text().catch(() => '');
+      return {
+        found: false,
+        retriable: false,
+        error: `Error calling read-only function. Response ${response.status}: ${response.statusText}. Attempted to fetch ${url} and failed with the message: "${msg}"`,
+      };
+    }
+    try {
+      const value = await response.json();
+      return { found: true, value: parseReadOnlyResponse(value) };
+    } catch {
```

Comment:
This empty `catch` statement without an error binding is strange. It looks like it was historically non-standard syntax, then was standardized in ES2019?
Regardless of if this is valid syntax, it indicates poor error handling. Errors should almost never be swallowed. In this case, it should at least log something useful.